CMakeLists.txt

```diff
@@ -27,7 +27,11 @@ list(APPEND CMAKE_MODULE_PATH ${CMAKE_SOURCE_DIR}/cmake)
 project(FleCSI LANGUAGES CXX C)
 
 set(CXX_STANDARD_REQUIRED ON)
-set(CMAKE_CXX_STANDARD 17)
+if(MSVC)
+  set(CMAKE_CXX_STANDARD 20)
```

Comment:
Why would we need C++20 but only for MSVC?

Reply:
This is not a language standard compliance issue. I had to enable C++20 support for MSVC as it makes the compiler itself more compliant (it wasn't able to properly compile certain constructs in C++17 mode). I should add a comment to the cmake file to clarify the rationale.

Reply:
I think you just need `/permissive-` to get the good behavior in any language mode, which would be preferable here merely because it conveys intent better.

Reply:
I'll try that, thanks. However, I think it will not be sufficient as the problems were caused by some premature template instantiation (required proper two-phase name lookup).

Reply:
I am certainly no MSVC expert, but I think two-phase lookup is one thing that that option _does_ in sufficiently recent versions of the compiler; its default value is different if you use a new enough language version.

Reply:
I tried this, setting `/permissive-` is not sufficient. I'll add a comment to the cmake file.